webrender/src/box_shadow.rs

```diff
@@ -88,7 +88,8 @@ impl FrameBuilder {
             if box_offset.x == 0.0 && box_offset.y == 0.0 && spread_amount == 0.0 {
                 return;
             }
-            let mut clips = Vec::new();
+            let mut clips = Vec::with_capacity(2);
+            clips.push(ClipSource::Rectangle(*prim_info.local_clip.clip_rect()));
```

Comment:
Not sure I understand this part. It looks like we are both adding the local clip rect to the clip list as well as the picture info and layer primitive info for box shadows. Sounds redundant. Please clarify.

Reply:
This is in a completely separate branch where using ClipSources *does* work for whatever reason.